### DISCUSSION TITLE
Find the name of instrument for each track

### DISCUSSION THREAD

**Discussion by jerryfan4** [mido/mido#533] [Q&A]
Hi,

Is there a way to obtain the name of instrument being used on different tracks?

**Comment by olemb** [comment]
Hi,

There is nothing built in to Mido but you can do something like this:

```
def get_intrument_name(track):
    for msg in track:
        if msg.type == 'instrument_name':
            return msg.name
    else:
        return ''
```

**Comment by mp-pinheiro** [comment]
> Hi,
> 
> There is nothing built in to Mido but you can do something like this:
> 
> ```
> def get_intrument_name(track):
>     for msg in track:
>         if msg.type == 'instrument_name':
>             return msg.name
>     else:
>         return ''
> ```

Hey! I know this is an old issue, sorry for reviving it, but for me it seems that this code only prints instruments when they are changed mid track. I ran the code using all tracks, but no message with 'instrument_name' type seems to exist:

```
for track in mid.tracks:
	for message in track:
		if message.type == 'instrument_name':
			print(message.type)
```

Is there a way of getting the instrument name of each track?

Thanks in advance!

**Comment by jjjjjjjjjjamiie** [comment]
> > Hi,
> > There is nothing built in to Mido but you can do something like this:
> > ```
> > def get_intrument_name(track):
> >     for msg in track:
> >         if msg.type == 'instrument_name':
> >             return msg.name
> >     else:
> >         return ''
> > ```
> 
> Hey! I know this is an old issue, sorry for reviving it, but for me it seems that this code only prints instruments when they are changed mid track. I ran the code using all tracks, but no message with 'instrument_name' type seems to exist:
> 
> ```
> for track in mid.tracks:
> 	for message in track:
> 		if message.type == 'instrument_name':
> 			print(message.type)
> ```
> 
> Is there a way of getting the instrument name of each track?
> 
> Thanks in advance!

The 'instrument_name' value in the example is a placeholder and should be replaced with the name of an instrument that you would like to search for. The tracks you're searching through will need to be labelled correctly for this to work.

**Comment by rdoursenaud** [comment]
I have a feeling we may be conflating two different ways of thinking about `instrument name` in the context of this exchange.

MIDI, as a protocol, has no concept of instrument names but has two very different concepts that can be misinterpreted as such:
- Program change, a message where a specific program number will make subsequent note messages to sound as the assigned instrument depending on the sound module used. There is an official specification that defines which instrument is attributed to which program for sound modules that comply with it: General MIDI
- Track label, a text string which only appear in Standard MIDI *Files* (And only of type 1 at that) and *may* be used to convey the intended playback sound to the user

The provided code sample seem to allude to the second possibility while its very possible that the question was framed with the first.

Please clarify accordingly.
